common/src/main/java/com/microsoft/identity/common/internal/dto/AccountCredentialBase.java

```diff
@@ -52,6 +54,21 @@ public void setAdditionalFields(Map<String, JsonElement> additionalFields) {
         mAdditionalFields = additionalFields;
     }
 
+    /**
+     * Adds the additionalFields of the input AccountCredentialBase to this instance.
+     *
+     * @param other The input AccountCredentialBase to merge with this instance.
+     */
+    public void mergeAdditionalFields(@NonNull final AccountCredentialBase other) {
+        if (null == mAdditionalFields) {
+            mAdditionalFields = new HashMap<>();
```

Comment:
If we're going to mutate this, this should really be a thread safe collection.

Reply:
Good call out. I've pushed a change in 14a16c8d658b76089fae071c1398a1808749226b with the goal of being API back-compat friendly (method sigs don't change)